From 9554c409308952f8c40b5b9fa72474931e7f90a7 Mon Sep 17 00:00:00 2001
From: Martin Kopec <mkopec@redhat.com>
Date: Wed, 25 Aug 2021 11:56:03 +0000
Subject: [PATCH 15/15] Reorder arguments and use parse from six

python3 doesn't care about the order of the args, however,
py2 does so set **params as the last argument in run_command
call.
The patch also import parse and request from six.moves.urllib.
---
 tempest/cmd/run.py                           | 8 ++++----
 tempest/scenario/test_dashboard_basic_ops.py | 4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/tempest/cmd/run.py b/tempest/cmd/run.py
index ae1bdb3e9..08dce120d 100644
--- a/tempest/cmd/run.py
+++ b/tempest/cmd/run.py
@@ -265,15 +265,15 @@ class TempestRun(command.Command):
             }
             try:
                 return_code = commands.run_command(
-                    **params, exclude_list=ex_list,
-                    include_list=in_list, exclude_regex=ex_regex)
+                    exclude_list=ex_list, include_list=in_list,
+                    exclude_regex=ex_regex, **params)
             except TypeError:
                 # exclude_list, include_list and exclude_regex are defined only
                 # in stestr >= 3.1.0, this except block catches the case when
                 # tempest is executed with an older stestr
                 return_code = commands.run_command(
-                    **params, blacklist_file=ex_list,
-                    whitelist_file=in_list, black_regex=ex_regex)
+                    blacklist_file=ex_list, whitelist_file=in_list,
+                    black_regex=ex_regex, **params)
             if return_code > 0:
                 sys.exit(return_code)
         return return_code
diff --git a/tempest/scenario/test_dashboard_basic_ops.py b/tempest/scenario/test_dashboard_basic_ops.py
index b1098fa9c..fdbdb3298 100644
--- a/tempest/scenario/test_dashboard_basic_ops.py
+++ b/tempest/scenario/test_dashboard_basic_ops.py
@@ -12,8 +12,8 @@
 
 import html.parser
 import ssl
-from urllib import parse
-from urllib import request
+from six.moves.urllib import parse
+from six.moves.urllib import request
 
 from tempest.common import utils
 from tempest import config
-- 
2.27.0

